InternxtDesktop/Services/BackupsService.swift

```diff
@@ -0,0 +1,114 @@
+//
+//  BackupsService.swift
+//  InternxtDesktop
+//
+//  Created by Richard Ascanio on 1/8/24.
+//
+
+import Foundation
+import RealmSwift
+
+class BackupsService: ObservableObject {
+    @Published var foldernames: [FoldernameToBackup] = []
+    @Published var urls: [URL] = []
+    @Published var isBackupButtonEnabled: Bool = false
+
+    private func getRealm() -> Realm {
+        do {
+            return try Realm(fileURL: ActivityManager.realmURL)
+        } catch {
+            error.reportToSentry()
+            fatalError("Unable to open Realm")
+        }
+
+    }
+
+    func clean() throws {
+        foldernames = []
+        urls = []
+        let realm = getRealm()
+        try realm.write {
+            realm.deleteAll()
+        }
+    }
+
+    func addFoldernameToBackup(_ foldernameToBackup: FoldernameToBackup) {
+        guard let url = URL(string: foldernameToBackup.url) else {
+            return
+        }
+
+        do {
+            let realm = getRealm()
+            try realm.write {
+                realm.add(foldernameToBackup)
+            }
+            self.urls.append(url)
+        } catch {
+            error.reportToSentry()
+        }
+    }
+
+    func removeFoldernameFromBackup(at index: Int) {
```

Comment:
don't use array indexes to find/remove items in a database, use an ID instead that the realm uses as primary key

https://stackoverflow.com/questions/64972993/how-i-can-get-new-unique-id-for-each-new-object-in-realm